services/navigation/builtin/builtin_test.go

```diff
@@ -154,56 +144,229 @@ func TestStartWaypoint(t *testing.T) {
 	)
 	test.That(t, err, test.ShouldBeNil)
 
-	pt := geo.NewPoint(1, 0)
-	err = ns.AddWaypoint(ctx, pt, nil)
-	test.That(t, err, test.ShouldBeNil)
-
-	pt = geo.NewPoint(3, 1)
-	err = ns.AddWaypoint(ctx, pt, nil)
-	test.That(t, err, test.ShouldBeNil)
-
-	err = ns.SetMode(ctx, navigation.ModeWaypoint, map[string]interface{}{"experimental": true})
-	test.That(t, err, test.ShouldBeNil)
-	ns.(*builtIn).activeBackgroundWorkers.Wait()
-
-	inputs, err := kinematicBase.CurrentInputs(ctx)
-	test.That(t, err, test.ShouldBeNil)
-	actualPt := geo.NewPoint(inputs[0].Value, inputs[1].Value)
-	test.That(t, actualPt.Lat(), test.ShouldEqual, pt.Lat())
-	test.That(t, actualPt.Lng(), test.ShouldEqual, pt.Lng())
-
-	// setup injected MoveOnGlobe to test what extra defaults to from startWaypointExperimental function
-	injectMS.MoveOnGlobeFunc = func(
-		ctx context.Context,
-		componentName resource.Name,
-		destination *geo.Point,
-		heading float64,
-		movementSensorName resource.Name,
-		obstacles []*spatialmath.GeoObstacle,
-		linearVelocity float64,
-		angularVelocity float64,
-		extra map[string]interface{},
-	) (bool, error) {
-		if extra != nil && extra["motion_profile"] != nil {
-			return true, nil
+	t.Run("Reach waypoints successfully", func(t *testing.T) {
+		injectMS.MoveOnGlobeFunc = func(
+			ctx context.Context,
+			componentName resource.Name,
+			destination *geo.Point,
+			heading float64,
+			movementSensorName resource.Name,
+			obstacles []*spatialmath.GeoObstacle,
+			linearVelocityMillisPerSec float64,
+			angularVelocityDegsPerSec float64,
+			extra map[string]interface{},
+		) (bool, error) {
+			err := kinematicBase.GoToInputs(ctx, referenceframe.FloatsToInputs([]float64{destination.Lat(), destination.Lng(), 0}))
+			return true, err
 		}
-		return false, errors.New("no motion_profile exist")
-	}
 
-	// construct new point to navigate to
-	pt = geo.NewPoint(0, 0)
-	err = ns.AddWaypoint(ctx, pt, nil)
-	test.That(t, err, test.ShouldBeNil)
-
-	err = ns.(*builtIn).startWaypointExperimental(map[string]interface{}{})
-	test.That(t, err, test.ShouldBeNil)
-	ns.(*builtIn).activeBackgroundWorkers.Wait()
```

Comment:
Moved this for the most part into a `t.Run` so I could group the tests and reuse the setup.